### ISSUE TITLE
Wrong visual hints

### ISSUE THREAD

**Issue by ankit-m** [OiWorld/MindTheWord#181]
I searched visual hints for the word **aus** (*out* in German). The visual hint led me to image search with the word *aus*. It showed pictures of Australia (aus -> australia).

This might happen to a lot of words which are in different language. Maybe we should append the language name after the word searched for. Ex. If *aus* is searched. We should search *aus German*. Although there is a very high chance that we get completely unrelated visual hints.

**Comment by Ceilican** [comment]
Yes...  That's an instance of a more general problem: the fact that Mind The Word always asks for translations (or visual hints) of words out of context.

I tried "aus German", as you suggested, and I got more images related to germany than to "aus".

I also tried "aus" in google.de, to check whether the german google would be smarter and know that "aus" is more probably related to "out" than to Australia, but I still got lots of images of Australia. Maybe that is because I am located in Australia, though.

I have also tried the source word "out" instead of the translated word "aus". Images of Australia disappeared, but of course there will be situations where the source word is ambiguous and the translated word is not (e.g. if we were doing the opposite translation, having "aus" as the source word and "out" as the translated word). However, I suspect that searching for visual hints of the source word might be slightly more precise. If we search for visual hints of the translated word, we have two sources of imprecision: the imprecision of the translation and the imprecision of the image search. If we search for visual hints of the source word, we have only one source of imprecision: that of the image search.

Also, I must say that the images I got for "out" were not very good either, because "out" is an abstract concept. In general, I think the visual hints feature will perform well only on concrete concepts.

I think this is an unsolvable problem in general. We cannot expect perfect visual hints. But we can do some small things to improve its accuracy. The idea of searching for source words might help. Maybe choosing the right localized version of Google (e.g. google.de instead of google.com) might have a small influence in certain cases, even though it didn't in the case of "aus".

In the meanwhile, if the visual hints are often misleading, I would recommend adding "(beta)" in the context menu title, in order to lower the user's expectation with respect to this feature. But only if *often*. We shouldn't worry too much about a single very particuylar special case such as "aus".
